### ISSUE TITLE
Freshly deployed apps have empty cf_app_name/cf_space_id/cf_space_name/etc

### ISSUE THREAD

**Issue by mrdavidlaing** [cloudfoundry-community/firehose-to-syslog#22]
When I deploy a new app, its firehose logs don't contain any data for cf_app_name/cf_space_id/cf_space_name/etc, ie:

```
{"cf_app_id":"653277ef-e7f3-4abb-93c6-c2295302f191","cf_app_name":"","cf_org_id":"","cf_org_name":"","cf_space_id":"","cf_space_name":"","event_type":"LogMessage","level":"info","message_type":"OUT","msg":"ruby-simple.apps.54.183.203.97.xip.io - [21/04/2015:06:42:45 +0000] \"GET / HTTP/1.1\" 200 811 \"-\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.104 Safari/537.36\" 10.10.2.6:37387 x_forwarded_for:\"80.229.7.108, 10.10.2.6\" vcap_request_id:1097e51b-8e38-4b49-4df5-7763e9490c46 response_time:0.005121271 app_id:653277ef-e7f3-4abb-93c6-c2295302f191\n","origin":"router__0","source_instance":"0","source_type":"RTR","time":"2015-04-21T06:42:28Z","timestamp":1429598565443002841}
```

If I restart the firehose-to-syslog service, then my new app's logs are correctly looked up, ie:
```
{"cf_app_id":"653277ef-e7f3-4abb-93c6-c2295302f191","cf_app_name":"ruby_simple","cf_org_id":"8005f45b-76d9-4038-8ca1-9e0a85ed5be0","cf_org_name":"system","cf_space_id":"a871a722-cad2-4d46-8061-2c9b728b7d8f","cf_space_name":"development","event_type":"LogMessage","level":"info","message_type":"OUT","msg":"ruby-simple.apps.54.183.203.97.xip.io - [21/04/2015:06:48:22 +0000] \"GET / HTTP/1.1\" 200 811 \"-\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.104 Safari/537.36\" 10.10.2.6:38632 x_forwarded_for:\"80.229.7.108, 10.10.2.6\" vcap_request_id:c22c2da7-d169-4551-7e8d-a6e2fc589946 response_time:0.004251063 app_id:653277ef-e7f3-4abb-93c6-c2295302f191\n","origin":"router__0","source_instance":"0","source_type":"RTR","time":"2015-04-21T06:48:05Z","timestamp":1429598902773388246}
```

I'm running the firehose-to-syslog service like this:
```
firehose-to-syslog \
        --domain=run.54.183.203.97.xip.io \
        --skip-ssl-validation  \
        --user=admin \
        --password=XXXXXXX \
        --syslog-server=127.0.0.1:5514 \
        --events=LogMessage
```

It appears that the `*Pull by application id if not cached yet` portion of the caching strategy isn't working.

**Comment by shinji62** [comment]
Strange work on my localhost, I will give a shot

**Comment by amhuber** [comment]
This is also working correctly for me with the latest develop code, the very first log entry for the new app (staging logs) has all the info filled in, and all subsequent app logs also look correct.

    {"cf_app_id":"f51b1cb7-4de5-4345-88f1-2bfb149b4125","cf_app_name":"amhtest","cf_org_id":"8a812e19-3665-4d8a-96ad-9129b3bc8ffc","cf_org_name":"intel","cf_space_id":"5c39e63b-7e37-43c6-b239-36d964545dff","cf_space_name":"Development","event_type":"LogMessage","level":"info","message_type":"OUT","msg":"-----\u003e Downloaded app package (4.0K)","origin":"legacy","source_instance":"3","source_type":"STG","time":"2015-04-21T22:23:23-07:00","timestamp":1429680203632978894}

**Comment by shinji62** [comment]
I find the pb I think I am making a PR

**Comment by shinji62** [comment]
PR done please checkout develop, you can remove the boltdb to be safe

**Comment by amhuber** [comment]
You might be commenting on the wrong issue, it looks like the PR will fix the problem in issue https://github.com/cloudfoundry-community/firehose-to-syslog/issues/21.  I'll comment over there as well but that did fix the crash.

**Comment by shinji62** [comment]
fix both normaly :)

**Comment by mrdavidlaing** [comment]
@shinji62 - thanks - I can confirm that this is fixed in my test environment by #23
